R/generateFilterValues.R

```diff
@@ -55,15 +55,16 @@ generateFilterValuesData = function(task, method = "randomForestSRC_importance",
 
   # ensemble
   if (class(method) == "list") {
-    ens.method = method[[1]]
-    method = method[[2]]
-    assertSubset(ens.method, choices = ls(.FilterEnsembleRegister), empty.ok = FALSE)
```

Comment:
Isn't this assertion still valid? We only want to have valid Ensemble filters as `ens.method`?

Reply:
The assertion is still valid but no longer needed because, in the code below,  we test to see if method[[1]] is in the FilterEnsembleRegister, to determine if this is an ensemble filter, then set ens.method to method[[1]]. Therefore ens.method is guaranteed to be in the FilterEnsembleRegister. 
```
  if (class(method) == "list") {
    if (method[[1]] %in% ls(.FilterEnsembleRegister)) {
	ens.method = method[[1]]
	method = method[[2]]
	if (length(method) == 1) {
	  warningf("You only passed one base filter method to an ensemble filter. Please use at least two base filter methods to have a voting effect.")
       }
    }
  }

```

Reply:
Ok. Can we replace the if clause with the proper assertion? 

https://github.com/mlr-org/mlr/blob/2a10e5a9a4a8a1ad3bff3501b05e07d423c76c21/R/generateFilterValues.R#L58

Reply:
I don't understand. Do you want:
 if (assertSubset(ens.method, choices = ls(.FilterEnsembleRegister), empty.ok = FALSE)) {

Reply:
Replacing

https://github.com/annette987/mlr/blob/541300f7b086a21d4a3787429f4a0359f8a98c34/R/generateFilterValues.R#L58

by 

`assertSubset(method[[1]], choices = ls(.FilterEnsembleRegister), empty.ok = FALSE)`

Then we simply throw an assertion error now if method[[1]] is not an ensemble method. We could be more specific with the error message here but this should suffice for now.

Reply:
No, that line needs to be an if statement. Previously, the base methods had to be a character vector of unique filter names. Now, we are allowing the base methods to be a named list. That meant having to delete this block of code:  
https://github.com/annette987/mlr/blob/e7ec04d99a857bf77b555ab03d2dcecadfc9ccc0/R/filterFeatures.R#L69-L73 
Note that it says "base.methods arrive here in a list when called from 'tuneParams'".
Therefore, we can no longer assume that the only methods passed as a list are ensemble methods. So we first check to see if method is of class list, and if it is we still need to check whether or not method[[1]] is an ensemble method. It may not be. If it is we do what we need to do for ensemble methods and if it is not we don't need to do anything. We can't simply make an assertion there.

Reply:
> we still need to check whether or not method[[1]] is an ensemble method. It may not be. If it is we do what we need to do for ensemble methods and if it is not we don't need to do anything. 

Now I see. With an assertion we would error which is not what we want in this case.
Thanks for resolving this in such detail!